lib/collection/src/operations/point_ops.rs

```diff
@@ -625,11 +678,20 @@ impl PointInsertOperationsInternal {
                     });
 
                     if let Some(payload) = point.payload {
-                        let set_payload = SetPayloadOp {
-                            points: Some(vec![point.id]),
-                            payload,
-                            filter: None,
-                            key: None,
+                        let set_payload = if let Some(update_filter) = update_filter.clone() {
+                            SetPayloadOp {
+                                points: None,
```

Comment:
Is `points: None` correct here?

Reply:
Yes it is correct. We can only set point IDs xor a filter, so we merge point IDs into the filter.